### ISSUE TITLE
"Open in IINA" is no longer support by safari 12.0

### ISSUE THREAD

**Issue by Bluetus** [lhc70000/iina#1988]
**System and IINA version:**

- macOS 10.13.6
- IINA 1.0.0-beta 4 Build 92

**Expected behavior:**

**Actual behavior:**

- Crash report:
```

Apple just delete the extension automatically after update safari 12.0
Doesn't allow to reinstall either

```

- Pic:
<img width="466" alt="screen shot 2018-09-18 at 3 38 13 am" src="https://user-images.githubusercontent.com/25444910/45672264-b4ed3b00-baf5-11e8-8b2d-6236b3e4bce4.png">

**Comment by LeoNatan** [comment]
@alexjs What does “not our problem” even means? Of course it’s your problem. As developers of IINA, part of the “contract” with the community is to support the latest technologies and constraints of the OS vendor.

**Comment by lhc70000** [comment]
Yes, surely it's our problem. 

We only need to submit the plugin to the Safari Extension Gallery. I'll get this done ASAP.

**Comment by LeoNatan** [comment]
@lhc70000 Is this the right approach? I think the gallery is being closed later this year. I think a better approach is to build an appex (extension) for Safari.

**Comment by LeoNatan** [comment]
> Support for .safariextz-style Safari Extensions installed from the Safari Extensions Gallery is deprecated with Safari 12 on macOS. Submissions to the Safari Extensions Gallery will no longer be accepted after December 2018. Developers are encouraged to transition to Safari App Extensions.

**Comment by AnanyaKirti** [comment]
Hope this happens!
Love IINA

**Comment by Bluetus** [comment]
Thank you, everyone! I confused when it was labeled as ”not our problem” too. Thought the extension was a different project somewhere else. Happy that this will happen. Looking forward to it! :)

**Comment by inflation** [comment]
@saagarjha Gladly

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by saagarjha** [comment]
After looking at #1990, I'm not quite sold on the need for a browser extension. Why can't we just vend a service so that it can be invoked from anywhere in macOS?

**Comment by AnanyaKirti** [comment]
yes, but I would still prefer an extension.
I mostly use IINA to watch 4k youtube videos in IINA (since I use Safari)

**Comment by lhc70000** [comment]
I think a service is only capable of executing small javascript code fragments. Since we may need to add more features related to browsers in the future, such as detecting and playing HTML5 videos on the page (and perhaps send cookies together to IINA/ytdl) and storing user preferences, I'd still prefer extensions which not only have more available APIs but also support browsers other than Safari.

BTW, @saagarjha, is there an estimated completion date?

**Comment by lhc70000** [comment]
Yes, definitely a system-wide service has its advantage and it's something that we should keep in mind in future development, but I think browser extensions are essential as well (and they're in urgent need right now).

**Comment by saagarjha** [comment]
Please take a look at #2029 and provide your feedback on this PoC.

**Comment by saagarjha** [comment]
Looks like this is merged. @lhc70000, I am keeping this open so you can implement the "other half" of this effort, namely, removing references to the old Safari extension, handling documentation for this, etc.
